packages/account/src/Sections/Security/TwoFactorAuthentication/digit-form.tsx

```diff
@@ -1,57 +1,68 @@
 import React from 'react';
 import classNames from 'classnames';
-import { Formik, Form, Field } from 'formik';
+import { Formik, Form, Field, FormikProps, FormikHelpers, FieldProps } from 'formik';
 import { Input, Button } from '@deriv/components';
 import { getPropertyValue, WS } from '@deriv/shared';
 import { localize } from '@deriv/translations';
+import { observer, useStore } from '@deriv/stores';
 
-const DigitForm = ({ is_enabled, setTwoFAStatus, setTwoFAChangedStatus, is_language_changing }) => {
+type TResponse = {
+    error?: {
+        message: string;
+        code: string;
+    };
+};
+
+type TDigitFormValues = {
+    digit_code: string;
+};
+
+const DigitForm = observer(() => {
+    const { client, common } = useStore();
+    const { is_language_changing } = common;
+    const { has_enabled_two_fa, setTwoFAChangedStatus, setTwoFAStatus } = client;
     const [is_success, setSuccess] = React.useState(false);
-    const [is_ready_for_verification, setReadyForVerification] = React.useState(false);
```

Comment:
why do we remove this usestate?

Reply:
This useState was unnecessarily complicating the validation logic. We can check if all OTP entered by user follows the validation simply by looking if formik error {} obj is empty before sending to backend. If BE responds with an error incase OTP is invalid, we will populate this error obj and disable the button.

Reply:
Reintroducing this state to make sure we send request for OTP validation only when user clicks btn. 🙏